src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/AdoNet/AdoNetExtensions.cs

```diff
@@ -0,0 +1,22 @@
+using System;
+using System.Runtime.CompilerServices;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AdoNet.SqlClient
+{
+    internal static class AdoNetExtensions
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static void DisposeWithException(this Scope scope, Exception exception)
+        {
+            if (scope != null)
+            {
+                if (exception != null)
+                {
+                    scope.Span.SetException(exception);
```

Comment:
Should we ensure the scope is finished even if `scope.Span.SetException(exception)` throws an exception?

Reply:
I wrapped this inside `try/finally` in 70318d8

Reply:
Thanks! One more thing 😅, do you think it's worth checking if `scope.Span` is null? The `try` would catch that, but it's nicer we can avoid the NRE.

Reply:
oh, I forgot that check, yeah, thanks for the catch.

Reply:
done in 9f7602f